app/js/lib/foxbox/settings.js

```diff
@@ -83,64 +76,24 @@ export default class Settings extends Model {
     storage.setItem(`${PREFIX}configured`, value);
   }
 
-  get localScheme() {
-    return this._localScheme;
-  }
-
-  set localScheme(scheme) {
-    scheme = String(scheme) || DEFAULT_SCHEME;
-    this._localScheme = scheme;
-    storage.setItem(`${PREFIX}localScheme`, this._localScheme);
-  }
-
-  get localHostname() {
-    return this._localHostname;
-  }
-
-  set localHostname(hostname) {
-    hostname = String(hostname) || DEFAULT_HOSTNAME;
-    this._localHostname = hostname.replace(/\/$/, ''); // Trailing slash.
-    storage.setItem(`${PREFIX}localHostname`, this._localHostname);
-  }
-
-  get localPort() {
-    return this._localPort;
-  }
-
-  set localPort(port) {
-    port = parseInt(port, 10) || DEFAULT_PORT;
-    this._localPort = port;
-    storage.setItem(`${PREFIX}localPort`, this._localPort);
-  }
-
-  get tunnelScheme() {
-    return this._tunnelScheme;
-  }
-
-  set tunnelScheme(scheme) {
-    scheme = String(scheme) || DEFAULT_SCHEME;
-    this._tunnelScheme = scheme;
-    storage.setItem(`${PREFIX}tunnelScheme`, this._tunnelScheme);
-  }
-
-  get tunnelHostname() {
-    return this._tunnelHostname;
+  get localUrl() {
+    return this._localUrl;
   }
 
-  set tunnelHostname(hostname) {
-    hostname = String(hostname) || DEFAULT_HOSTNAME;
-    this._tunnelHostname = hostname.replace(/\/$/, ''); // Trailing slash.
-    storage.setItem(`${PREFIX}tunnelHostname`, this._tunnelHostname);
+  set localUrl(url) {
+    url = String(url) || DEFAULT_URL;
+    this._localUrl = url.replace(/\/$/, ''); // Trailing slash.
```

Comment:
If we agree that it's ```origin``` and not ```url``` we can do just ```this._localOrigin = url ? (new URL(url)).origin : DEFAULT_ORIGIN;``` that will guard us from non-valid URL's (at least at development stage) and get rid of trailing slashes :)